ui/components/DashboardComponent.js

```diff
@@ -238,92 +240,94 @@ class DashboardComponent extends React.Component {
   fetchMetricComponents = () => {
     const self = this;
 
-    dataFetch(
-      "/api/telemetry/metrics/config",
-      {
-        method: "GET",
-        credentials: "include",
-        headers: {
-          "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8",
+    if(this.state.isMeshConfigured){
```

Comment:
it's better to conditionally *call* this function based on the value of `isMeshConfigured`.